### PR TITLE
rm falsy

### PR DESCRIPTION
**Pull request by maelle**

### CODE REVIEW

File: R/update.r [hunk 1]
```diff
@@ -189,28 +192,28 @@ add_versions <- function(object, to_add, archive, current) {
...
-    paste(sep = "/",
-          cran_site(),
-          paste(archive_path_comps, collapse = "/"),
-          tarnames)
+  
+  if (length(tarnames) > 0) {
+    url1 <- paste(sep = "/",
+                  cran_site(),
+                  paste(archive_path_comps, collapse = "/"),
+                  tarnames)
```
Comment (maelle) [inline R/update.r]: @gaborcsardi above should I add tarnames inside paste as well? I think I might have missed what `&%%&` did.

File: R/update.r [hunk 2]
```diff
@@ -189,28 +192,28 @@ add_versions <- function(object, to_add, archive, current) {
...
-          tarname2)
+                                   paste0(pkg, "_", versions, ".tar.gz")]
+  
+  if (length(tarname2) > 0) {
+    
+    url2 <- paste(sep = "/",
+                  cran_site(),
+                  paste(pkg_path_comps, collapse = "/"),
+                  tarname2)
   }
```
Comment (maelle) [inline R/update.r]: same doubt here

**Comment by maelle** [comment]
The tests are failing so I must have misinterpreted part of falsy functionality.

**Comment by gaborcsardi** [comment]
Another way around this would be to just add a `falsy.R` file with all the falsy code. Although that might conflict with pkgsearch when we merge....

**Comment by maelle** [comment]
Well I did add a replacement for `%||%`, was `falsy`'s one different?

`%&&%` seemed more puzzling hence my trying to replace it. But after all maybe the errors come from `%||%`?!

**Comment by gaborcsardi** [comment]
Yeah, different:
```
❯ falsy::is_falsy
function (object)
{
    is.null(object) || identical(object, FALSE) || identical(object,
        0L) || identical(object, 0) || identical(object, 0 +
        (0+0i)) || identical(object, "") || identical(object,
        as.raw(0)) || identical(object, logical()) || identical(object,
        integer()) || identical(object, double()) || identical(object,
        complex()) || identical(object, character()) || identical(object,
        raw()) || identical(object, list()) || inherits(object,
        "try-error")
}
```

**Comment by gaborcsardi** [comment]
But it turns out this is not a good idea in this language, because of the vectorization, that's why it was deprecated.

**Comment by maelle** [comment]
So it'd be better to adapt the code to use the current version of `%||%`? Could you have a look?

File: R/update.r (hunk 1 above)
Comment (gaborcsardi) [reply R/update.r]: no, its good I think

**Comment by gaborcsardi** [comment]
Thanks!

**Pull request merged**